src/node/node_main.ml

```diff
@@ -551,6 +560,54 @@ let _main_2 (type s)
           and ncnum = cluster_cfg.ncnum
           in
           S.make_store ~lcnum ~ncnum db_name >>= fun (store:S.t) ->
+          Lwt.return (tlog_coll, store)
+        in
+        Lwt.catch
+        (fun () -> _open_tlc_and_store ())
+        (fun exn ->
+         if autofix
+         then
+           begin
+             let clean_up_retry fn =
+               (if not !head_copied
+               then File_system.unlink db_name
+               else Lwt.return_unit)
+               >>= fun () ->
+               Tlog_main._mark_tlog fn (Tlc2._make_close_marker me.node_name) >>= fun _ ->
+               _open_tlc_and_store()
+             in
+             match exn with
+             | Tlc2.TLCNotProperlyClosed _ ->
+                begin
+                  Logger.warning_ ~exn "AUTOFIX: improperly closed tlog"
+                  >>= fun ()->
+                  Tlc2.get_last_tlog me.tlog_dir me.tlx_dir
+                  >>= fun (_, fn) ->
+                  clean_up_retry fn
+                end
+             | Tlogcommon.TLogUnexpectedEndOfFile _ ->
+                begin
+                  Logger.warning_ ~exn "AUTOFIX: unexpected end of tlog"
+                  >>= fun () ->
+                  Tlc2.get_last_tlog me.tlog_dir me.tlx_dir
+                  >>= fun (_, fn) ->
+                  Tlc2._truncate_tlog fn >>= fun ()->
```

Comment:
I would've chosen to truncate the tlog after removing the store ... but I can't really make a hard case for when why it would be better. Just feels more right to always first remove the store and then do the fixup.